devp2p/discovery.py

```diff
@@ -218,7 +222,7 @@ def __init__(self, app, transport):
 
     def get_node(self, nodeid, address=None):
         "return node or create new, update address if supplied"
-        assert isinstance(nodeid, str)
+        assert isinstance(nodeid, bytes)
         assert len(nodeid) == 512 / 8
```

Comment:
Yep, I did not change it because the result will always be a whole number (where float == int). Will fix it.